src/prediction.jl

```diff
@@ -185,8 +184,40 @@ function check_longdata_metaduplicates!(longdata_df::DataFrame; remove_duplicate
     end # end if n_unique_rows
 end # end function
 
-function build_future_df(base_df, to_predict::Symbol)
+function build_metadata_prediction_df(base_df, inputs::Vector{Symbol}, targets::Vector{Symbol})
 
+    subjects = unique(base_df.subject)
+    has_stool = findall(.!(ismissing.(base_df.sample)))
+    has_cogscore = findall(.!(ismissing.(base_df.cogScore)))
+
+    subjects_stool_idx = [ intersect(findall(base_df.subject .== el), has_stool) for el in subjects ]
+    subjects_cogscore_idx = [ intersect(findall(base_df.subject .== el), has_cogscore) for el in subjects ]
+
+    line_combinations = vcat( [ vec(collect(Base.product(subjects_stool_idx[i], subjects_cogscore_idx[i]))) for i in 1:length(subjects) ]... )
+    line_combinations = line_combinations[ [ el[2] > el[1] for el in line_combinations ] ]
```

Comment:
It feels like there must be a better way of doing this. But I'm not totally sure... I'll keep thinking about it.

Reply:
This version is faster than the previous one, though (the `build_future_df`)